### ISSUE TITLE
Design and Implement Validator Attestation LMD Pool

### ISSUE THREAD

**Issue by terenc3t** [prysmaticlabs/prysm#1322]
The expectation of a validator is to run a node that is listening for both attestations and blocks on the wire. The default behavior of a beacon chain full node should be to support listening to attestations on the wire and reacting to incoming validator's latest attestation to determine fork choice. As a validator, I will track each validator latest vote. As a proof of concept, the attestation pool can look like the following:

```
{
    validator1PubKey: latestAttestationData
    validator2PubKey: latestAttestationData
    validator3PubKey: latestAttestationData
    validator4PubKey: latestAttestationData
...
}
```

**Comment by terenc3t** [comment]
After speaking with @prestonvanloon, we both agree it's fine for this attestation pool to be in memory (not disk)

**Comment by terenc3t** [comment]
Closed via #1402
